test/signalrclient-e2e-tests/connection_tests.cpp

```diff
@@ -0,0 +1,106 @@
+// Copyright (c) Microsoft Open Technologies, Inc. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+#pragma once
+
+#include "stdafx.h"
+#include <string>
+#include "cpprest\details\basic_types.h"
+#include "cpprest\json.h" 
+#include "connection.h"
+#include "hub_connection.h"
+#include "url.h"
+
+TEST(connection_tests, connection_status_start_stop)
+{
+    auto conn = std::make_shared<signalr::connection>(url + U("raw-connection"));
+    auto test_event = std::make_shared<pplx::event>();
+
+    try
+    {
+        conn->start().get();
+        ASSERT_EQ(conn->get_connection_state(), signalr::connection_state::connected);
+        conn->stop().get();
+        ASSERT_EQ(conn->get_connection_state(), signalr::connection_state::disconnected);
+        conn->start().get();
+        ASSERT_EQ(conn->get_connection_state(), signalr::connection_state::connected);
+        test_event->set();
+    }
+    catch (...)
+    {
+        ASSERT_TRUE(false);
+    }
+
+    ASSERT_FALSE(test_event->wait(15000));
+}
+
+TEST(connection_tests, send_message)
+{
+    auto conn = std::make_shared<signalr::connection>(url + U("raw-connection"));
+    auto message = std::make_shared<utility::string_t>();
+    auto test_event = std::make_shared<pplx::event>();
+
+    conn->set_message_received([message, test_event](const utility::string_t& payload)
+    {
+        *message = payload;
+        test_event->set();
+    });
+
+    try
+    {
+        conn->start().then([conn]()
+        {
+            web::json::value obj;
+            obj[U("type")] = web::json::value::number(0);
+            obj[U("value")] = web::json::value::string(U("test"));
+            return conn->send(obj.serialize());
+
+        }).get();
+    }
+    catch (...)
+    {
+        ASSERT_TRUE(false);
```

Comment:
You may want to catch `const std::exception& e` here instead of `...` and write the exception to the `ucout`. Otherwise if this test fails we will not have any idea why it failed. Yes it would mess the CI but only in the case where the run would have failed anyways so it should be fine - alternatively you may just remove the try/catch block. You are now doing `.get` so the exception is being observed (i.e. the process won't crash) and `gtest` will catch it and report (including the message). I think we should do the latter everywhere as long as we are sure that exceptions are being observed.